coderd/telemetry/telemetry.go

```diff
@@ -0,0 +1,585 @@
+package telemetry
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"runtime"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/elastic/go-sysinfo"
+	"github.com/google/uuid"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+
+	"github.com/coder/coder/buildinfo"
+	"github.com/coder/coder/coderd/database"
+)
+
+type Options struct {
+	Database database.Store
+	Logger   slog.Logger
+	URL      *url.URL
```

Comment:
This is a bit ambiguous, maybe naming it `TelemetryURL` or a comment would be helpful.